src/likelihoodratiotest.jl

```diff
@@ -221,20 +221,40 @@ function StatsModels.isnested(m1::MixedModel, m2::MixedModel; atol::Real=0.0)
     end || return false
 
     # check that the nested fixef are a subset of the outer
-    all(in.(coefnames(m1), Ref(coefnames(m2)))) || return false
+    # XXX: the simple coefficient name check is nice and fast, but fails
+    # if model uses a pre-computed centered value and another uses
+    # StandardizedPredictors (an example of which we have in EmbraceUncertainty,
+    # where we use StandardizedPredictors in the simpler model, but swap
+    # to pre-computation for a more complicated model with a quadratic term)
+    # all(in.(coefnames(m1), Ref(coefnames(m2)))) || return false
+    if !_isnested(modelmatrix(m1), modelmatrix(m2))
+        @error "Fixed effects are not nested."
+        return false
+    end
 
     # check that the same grouping vars occur in the outer model
     grpng1 = fname.(m1.reterms)
     grpng2 = fname.(m2.reterms)
 
-    all(in.(grpng1, Ref(grpng2))) || return false
-
+    if !isempty(setdiff(grpng1, grpng2))
+        @error "Inner models have grouping variables not present in outer models."
+        return false
+    end
     # check that every intercept/slope for a grouping var occurs in the
```

Comment:
Is the reason for switching to `setdiff` just for clarity?

Also, both here and above there is a call to `@error` followed by `return false`.  My guess is that the `return false` will never be called as `@error` does not return.  I have no objections to this formulation but I am wondering if it is redundant.